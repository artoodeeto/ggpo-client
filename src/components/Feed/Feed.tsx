import React, { useEffect, FC } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import PostContainer from 'components/Post/PostContainer';
import { IPost } from 'interfaces/post';
import { querySomePost } from 'store/feedPost/Actions';
import { posts } from 'store/feedPost/Selectors';
// import { isUserAuthorized } from 'store/session/Selectors';

type FeedProps = {
  // queryPostsForFeed: (offset: number, limit: number) => void;
  // posts: IPost[];
};

const Feed: FC<FeedProps> = () => {
  const dispatch = useDispatch();
  const postsFeed: IPost[] = useSelector(posts);
  useEffect(() => {
    // queryPostsForFeed(0, 5);
    dispatch(querySomePost(0, 5));
  }, [dispatch]);
  // TODO: load new posts when scrolling
  return (
    <div>
      <Link to="/profile">Profile</Link>
      <br />
      <Link to="/game_groups">GG</Link>

      <h1>I am FEED TEST!</h1>
      <PostContainer posts={postsFeed} showOptionsBtn={false} />
    </div>
  );
};

// const mapStateToProps = (state: IState) => {
//   return {
//     isAuthenticated: isUserAuthorized(state),
//     posts: posts(state)
//   };
// };

// const mapDispatchToProps = (dispatch: Function) => {
//   return {
//     queryPostsForFeed: (offset: number, limit: number) => dispatch(querySomePost(offset, limit))
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(Feed);

export default Feed;
