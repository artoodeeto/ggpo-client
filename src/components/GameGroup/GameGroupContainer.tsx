import React, { Component, Fragment } from 'react';
import { IState } from 'interfaces/stateInterface';
import { connect } from 'react-redux';
import GameGroupList from './GameGroupList/GameGroupList';

class GameGroupContainer extends Component<any> {
  render() {
    return (
      <Fragment>
        <h1>GameGroups</h1>
        <GameGroupList />
      </Fragment>
    );
  }
}

const mapStateToProps = (state: IState) => {
  return {};
};

const mapDispatchToProps = (dispatch: Function) => {
  return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(GameGroupContainer);
