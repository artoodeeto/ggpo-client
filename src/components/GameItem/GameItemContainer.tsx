import React, { Component } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { getAGameGroupAPI } from 'api/gamegroups/gamegroups';
import { connect } from 'react-redux';
import { setIsFollower } from 'store/gameGroupItem/Actions';
import { IGameGroup } from 'interfaces/gameGroup';
import GameItem from './GameItem/GameItem';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { RootState } from 'store/root/root_reducer';

interface RouteCustomParams {
  id: string;
}

export interface GameItemContainerProps extends RouteComponentProps<RouteCustomParams> {
  setIsFollower: (isFollowing: boolean) => void;
  isFollowing: boolean;
}

export interface GameItemContainerState {
  isFollower: boolean;
  theGameGroup: IGameGroup;
}

class GameItemContainer extends Component<GameItemContainerProps, GameItemContainerState> {
  constructor(props: GameItemContainerProps) {
    super(props);
    this.state = {
      isFollower: false,
      theGameGroup: {
        id: 0,
        description: '',
        title: '',
        createdAt: '',
        updatedAt: ''
      }
    };
  }

  async componentDidMount() {
    const { id } = this.props.match.params;
    try {
      const { payload } = await getAGameGroupAPI(Number(id));
      const { isFollower, gameGroup } = payload;
      this.setState({ isFollower, theGameGroup: gameGroup });
      this.props.setIsFollower(isFollower);
    } catch (error) {
      // TODO: add error handling here
      console.log(error);
    }
  }

  render() {
    // the background color here should change if the game is "hot" like people are tagging this game or talking about it. something like that lols
    return (
      <div>
        <GameItem isFollower={this.props.isFollowing} gameItem={this.state.theGameGroup} />
      </div>
    );
  }
}

const mapStateToProps = (state: RootState) => {
  return {
    isFollowing: state.gameGroupItem.gameGroupItem.isFollower
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    setIsFollower: (isFollowing: boolean) => dispatch(setIsFollower(isFollowing))
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GameItemContainer));
