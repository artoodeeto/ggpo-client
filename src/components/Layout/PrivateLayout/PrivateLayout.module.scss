@import '../../../assets/media-queries.scss';

.LeftAside {
  @include sm {
    display: none;
  }

  @include md {
    display: none;
  }
}

.RightAside {
  @include sm {
    display: none;
  }

  @include md {
    display: none;
  }
}
