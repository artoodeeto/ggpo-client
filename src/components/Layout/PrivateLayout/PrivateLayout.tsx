import React, { FC } from 'react';

type PrivateLayoutProps = {};

const PrivateLayout: FC<PrivateLayoutProps> = ({ children }) => {
  return (
    <section>
      <aside></aside>
      <main>{children}</main>
      <aside></aside>
    </section>
  );
};

export default PrivateLayout;
