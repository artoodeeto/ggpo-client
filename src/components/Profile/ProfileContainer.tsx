import React, { Component } from 'react';
import { connect } from 'react-redux';
import Cookies from 'js-cookie';
import PostForm from 'components/shared/PostForm/PostForm';
import PostContainer from 'components/Post/PostContainer';
import { getUserPosts } from 'store/userProfilePost/Actions';
import { userSelectorUserId } from 'store/user/Selectors';
import { userProfileSelectorPosts } from 'store/userProfilePost/Selectors';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { RootState } from 'store/root/root_reducer';
import { IPost } from 'interfaces/post';
import { logoutSession } from 'store/session/Actions';
import { logoutUser } from 'store/user/Actions';

export interface ProfileContainerProps {
  currentUserId: number;
  posts: IPost[];
  clearSession: () => void;
  clearUser: () => void;
  userPosts: (userID: number, offset: number, limit: number) => void;
}

export interface ProfileContainerState {}

class ProfileContainer extends Component<ProfileContainerProps, ProfileContainerState> {
  componentDidMount() {
    this.props.userPosts(this.props.currentUserId, 0, 5);
  }

  logOutCurrentUser() {
    this.props.clearSession();
    this.props.clearUser();
    Cookies.remove(process.env.REACT_APP_COOKIE_NAME as string);
  }

  render() {
    return (
      <div>
        <button type="button" onClick={() => this.logOutCurrentUser()}>
          Log out
        </button>
        <h1>PROFILE</h1>
        <PostForm />
        <PostContainer posts={this.props.posts} showOptionsBtn />
      </div>
    );
  }
}

const mapStateToProps = (state: RootState) => {
  return {
    currentUserId: userSelectorUserId(state),
    posts: userProfileSelectorPosts(state)
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    clearSession: () => dispatch(logoutSession()),
    clearUser: () => dispatch(logoutUser()),
    userPosts: (userID: number, offset: number, limit: number) => dispatch(getUserPosts(userID, offset, limit))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfileContainer);
