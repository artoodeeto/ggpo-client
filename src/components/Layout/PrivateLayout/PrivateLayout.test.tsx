import { shallow } from 'enzyme';
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Feed from 'components/Feed/Feed';
import ProfileContainer from 'components/Profile/ProfileContainer';
import GameGroupContainer from 'components/GameGroup/GameGroupContainer';
import GameItemContainer from 'components/GameItem/GameItemContainer';
import PrivateLayout from './PrivateLayout';

describe('<PrivateLayout/>', () => {
  describe('PrivateLayout snapshots', () => {
    it('user is on login or signup page', () => {
      const wrapper = shallow(
        <PrivateLayout>
          <Feed />
        </PrivateLayout>
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('user is on profile page', () => {
      const wrapper = shallow(
        <PrivateLayout>
          <ProfileContainer />
        </PrivateLayout>
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('user is on gamegroup page', () => {
      const wrapper = shallow(
        <PrivateLayout>
          <GameGroupContainer />
        </PrivateLayout>
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('user is on gamegroup page', () => {
      const wrapper = shallow(
        <PrivateLayout>
          <GameItemContainer />
        </PrivateLayout>
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
