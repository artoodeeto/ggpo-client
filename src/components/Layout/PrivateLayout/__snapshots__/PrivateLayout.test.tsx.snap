// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateLayout/> PrivateLayout snapshots user is on gamegroup page 1`] = `
<section>
  <aside />
  <main>
    <Connect(GameGroupContainer) />
  </main>
  <aside />
</section>
`;

exports[`<PrivateLayout/> PrivateLayout snapshots user is on gamegroup page 2`] = `
<section>
  <aside />
  <main>
    <withRouter(Connect(GameItemContainer)) />
  </main>
  <aside />
</section>
`;

exports[`<PrivateLayout/> PrivateLayout snapshots user is on login or signup page 1`] = `
<section>
  <aside />
  <main>
    <Feed />
  </main>
  <aside />
</section>
`;

exports[`<PrivateLayout/> PrivateLayout snapshots user is on profile page 1`] = `
<section>
  <aside />
  <main>
    <Connect(ProfileContainer) />
  </main>
  <aside />
</section>
`;
